### PR TITLE
fix(Drawer): Add onToggle callback

### PR DESCRIPTION
**Pull request by maroun-baydoun**
**What kind of change does this PR introduce?** (check at least one)

This PR adds an `onToggle` callback prop. This will be used to let the outside world know when the drawer is being toggled.

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] All tests are passing
- [x] New/updated tests are included
- [x] Run the precommit hook (it's going to run all required actions such as `Prettier`, `Eslint`, `Jest` and so on...)

**Other information:**

### CODE REVIEW

File: src/components/Drawer/tests/Drawer.test.js [hunk 1]
```diff
@@ -98,4 +100,20 @@ describe('<Drawer />', () => {
     component.unmount();
     expect(unbindEscKeyFn.mock.calls.length).toBe(1);
   });
+
+  it('should call onToggle with a true param when the trawer is visible', () => {
```
Comment (jarjan) [inline src/components/Drawer/tests/Drawer.test.js]:
```suggestion
  it('should call onToggle with a true param when the drawer is visible', () => {
```

File: src/components/Drawer/tests/Drawer.test.js [hunk 2]
```diff
@@ -98,4 +100,20 @@ describe('<Drawer />', () => {
...
+    const onToggle = jest.fn();
+    const component = renderedComponent({ onToggle });
+
+    component.setProps({ open: true });
+    expect(onToggle).toBeCalledWith(true);
+  });
+
+  it('should call onToggle with a false param when the trawer is hidden', () => {
```
Comment (jarjan) [inline src/components/Drawer/tests/Drawer.test.js]:
```suggestion
  it('should call onToggle with a false param when the drawer is hidden', () => {
```

File: src/components/Drawer/tests/Drawer.test.js (hunk 1 above)
Comment (maroun-baydoun) [reply src/components/Drawer/tests/Drawer.test.js]: Thank you for catching that!

**Pull request merged**